### DISCUSSION TITLE
Info on the `npm outdated` command

### DISCUSSION THREAD

**Discussion by vizzale** [npm/feedback#290] [General]
Hi, first for this kind of questions I used the forum, I hope this is therefore the right place, case opposite tell me where to write. If I launch the `npm outdated` command, after accessing my project's folder, this is the situation:

![screen](https://user-images.githubusercontent.com/37701762/111765698-fae11080-88a4-11eb-9405-cb6a988448dc.png)

I am not an expert and consulting site resources I only understand that these two packages are not updated to the latest version available. What can it depend on? Tell me if I wasn't clear and you need other elements.
Thanks in advance.

**Comment by ljharb** [comment]
@vizzale What's happening here is that the jump to the latest version from the "current" and/or "wanted" versions is a semver-major one, meaning there's breaking changes. Generally the package that directly depends on them will need to be manually updated to those versions, and the breaking changes accounted for.

**Reply by vizzale** [reply]
@ljharb And what do you advise me to do?

**Reply by ljharb** [reply]
If you directly depend on `postcss-sorting`, your choices are:
1. do nothing, and keep using v5. if it works, it won't stop working.
1. `npm install --save-dev postcss-sorting@latest` and update to v6. Read the changelog carefully and update your code as needed to account for the breaking changes. Repeat for `stylelint-config-rational-order` and any other deps in this situation.

**Reply by MylesBorins** [reply]
@vizzale you may also want to check `npm audit` to see if there are any known security issues.

While it is generally good hygiene to keep your dependencies up to date, you could break your application by updating them. It is up to you to determine your risk tolerance as well as ensure you have adequate testing in place to ensure updating dependencies won't break anything.

**Reply by vizzale** [reply]
Hi guys, I waited a while to add an answer because first I wanted to carry out the tests independently, avoiding as much as possible to steal your time. My use of Node.js and npm is quite limited for now, so too is my knowledge of them. In fact, I use them to install `stylelint`,` stylelint-order`, `stylelint-config-rational-order`,` postcss`, `postcss-sorting`.

I followed the first advice and installed the latest version (@latest) of all 5 devDependences using the `npm install` command. Now they are all up to date and the linter is working properly again, so thanks @ljharb.

Now I have two questions:
1. Why was it necessary to specify to npm to install the latest version? Shouldn't he have done it automatically?
2. I also ran the `npm audit` command as suggested by @MylesBorins and this is the report generated:

```
yargs-parser  <=13.1.1 || 14.0.0 - 15.0.0 || 16.0.0 - 18.1.1
Prototype Pollution - https://npmjs.com/advisories/1500
fix available via `npm audit fix --force`
Will install stylelint-config-rational-order@0.0.4, which is a breaking change
node_modules/stylelint-config-rational-order/node_modules/yargs-parser
  meow  5.0.0 - 6.0.1
  Depends on vulnerable versions of yargs-parser
  node_modules/stylelint-config-rational-order/node_modules/meow
    stylelint  9.2.1 - 12.0.1
    Depends on vulnerable versions of meow
    node_modules/stylelint-config-rational-order/node_modules/stylelint
      stylelint-config-rational-order  >=0.0.4
      Depends on vulnerable versions of stylelint
      Depends on vulnerable versions of stylelint-order
      node_modules/stylelint-config-rational-order
      stylelint-order  2.0.0 - 3.1.0
      Depends on vulnerable versions of stylelint
      node_modules/stylelint-config-rational-order/node_modules/stylelint-order

5 low severity vulnerabilities

To address all issues (including breaking changes), run:
npm audit fix --force 
```

What do you think?
